runelite-client/src/main/java/net/runelite/client/plugins/farmingtracker/FarmingTrackerPlugin.java

```diff
@@ -0,0 +1,413 @@
+
+		//Reload the config on enabling the plugin
+		if (client != null && client.getGameState() == GameState.LOGGED_IN)
+		{
+			loadConfig();
+		}
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		lastUsername = null;
+		pluginToolbar.removeNavigation(navButton);
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked menuOptionClicked)
+	{
+		if ((menuOptionClicked.getMenuAction() == MenuAction.GAME_OBJECT_FIRST_OPTION && firstOptions.contains(menuOptionClicked.getMenuOption())) || menuOptionClicked.getMenuAction() == MenuAction.ITEM_USE_ON_GAME_OBJECT)
+		{
+			PatchLocation patchLocation = PatchLocation.findByWorldLocation(client, queryRunner, menuOptionClicked.getId() & 127, menuOptionClicked.getId() >> 7 & 127);
+
+			if (patchLocation != null)
+			{
+				clickedOnPatch = patchLocation;
+
+				if (patchLocation.equals(PatchLocation.FOSSIL_ISLAND_TREE_HARDWOOD_EAST)
+					|| patchLocation.equals(PatchLocation.FOSSIL_ISLAND_TREE_HARDWOOD_MIDDLE)
+					|| patchLocation.equals(PatchLocation.FOSSIL_ISLAND_TREE_HARDWOOD_WEST))
+				{
+					menuTargetHardWood = Text.removeTags(menuOptionClicked.getMenuTarget());
+				}
+			}
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		String message = event.getMessage();
+
+		if (event.getType() == ChatMessageType.SERVER && message.equals("The herb patch is now empty.") || message.equals("The hops patch is now empty."))
+		{
+			handlePatchCleared();
+		}
+
+		if (event.getType() != ChatMessageType.FILTERED)
+		{
+			return;
+		}
+
+		if (message.equals("You plant the plantpot in the hardwood tree patch."))
+		{
+			handlePlantedSaplingHardwood();
+		}
+		else if (message.startsWith(CHAT_MESSAGE_PLANTED_SEED))
+		{
+			handlePlantedSeed(message);
+		}
+		else if (message.equals("You have successfully cleared this patch for new crops.")
+			|| message.equals("The allotment is now empty.")
+			|| message.equals("You dig up the tree stump."))
+		{
+			handlePatchCleared();
+		}
+	}
+
+	@Subscribe
+	public void onAnimationChanged(AnimationChanged event)
+	{
+		if (event.getActor() != client.getLocalPlayer() || clickedOnPatch == null)
+		{
+			return;
+		}
+
+		ObjectComposition currentPatch = client.getObjectDefinition(clickedOnPatch.getPatchObjectId()).getImpostor();
+
+		switch (event.getActor().getAnimation())
+		{
+			case FARMING_HARVEST_FLOWER:
+				previousPatchObject = currentPatch;
+				previousAnimation = FARMING_HARVEST_FLOWER;
+				break;
+			case IDLE:
+				handleHarvest(currentPatch);
+				break;
+		}
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (event.getGameState() != GameState.LOGIN_SCREEN && event.getGameState() == GameState.LOGGED_IN && !client.getUsername().equals(lastUsername))
+		{
+			lastUsername = client.getUsername();
+
+			loadConfig();
+		}
+	}
+
+	@Schedule(
+		period = 600,
+		unit = ChronoUnit.MILLIS
+	)
+	public void runFarmingTickChecker()
```

Comment:
Cannot this be replaced with GameTick hook? Or is it required for this to be run also in login screen?